### PR TITLE
Add password text field with eye toggle

### PR DESCRIPTION
**Pull request by AhmedAtef277**

### CODE REVIEW

File: Shared_UI/Shared_UI.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -148,7 +154,7 @@
 					};
 				};
 			};
-			buildConfigurationList = B62D386529BA635400179020 /* Build configuration list for PBXProject "Shared_UI" */;
+			buildConfigurationList = B62D386529BA635400179020 /* Build configuration list for PBXProject "shared_UI" */;
```
Comment (Mahmoud-Bassuni) [inline Shared_UI/Shared_UI.xcodeproj/project.pbxproj:157]: why this change

File: Shared_UI/Shared_UI.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -401,7 +409,7 @@
 /* End XCBuildConfiguration section */
 
 /* Begin XCConfigurationList section */
-		B62D386529BA635400179020 /* Build configuration list for PBXProject "Shared_UI" */ = {
+		B62D386529BA635400179020 /* Build configuration list for PBXProject "shared_UI" */ = {
```
Comment (Mahmoud-Bassuni) [inline Shared_UI/Shared_UI.xcodeproj/project.pbxproj:412]: the same

File: Shared_UI/Shared_UI.xcodeproj/project.pbxproj (hunk 1 above)
Comment (AhmedAtef277) [reply Shared_UI/Shared_UI.xcodeproj/project.pbxproj:157]: i didn't make this change! , how to fix it

File: Shared_UI/Shared_UI/Extension/UITextField + Extension.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    private func createToggleButton() -> UIButton {
+        let eyeButton = UIButton()
+        eyeButton.setImage(.hidePassswordIcon, for: .normal)
+        eyeButton.tintColor = .black
+        if #available(iOS 15.0, *) {
+            var eyeButtonConfiguration = UIButton.Configuration.plain()
+            eyeButtonConfiguration.buttonSize = .small
+            eyeButton.configuration = eyeButtonConfiguration
+        } else {
+            eyeButton.imageEdgeInsets = UIEdgeInsets(top: 0, left: -8, bottom: 0, right: 8)
```
Comment (Mahmoud-Bassuni) [inline Shared_UI/Shared_UI/Extension/UITextField + Extension.swift:27]: why those hard-coded value like 8 , -8

File: Shared_UI/Shared_UI.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Mahmoud-Bassuni) [reply Shared_UI/Shared_UI.xcodeproj/project.pbxproj:157]: you can change on this file manual

File: Shared_UI/Shared_UI/Extension/UITextField + Extension.swift (hunk 3 above)
Comment (AhmedAtef277) [reply Shared_UI/Shared_UI/Extension/UITextField + Extension.swift:27]: 8 to give space from the right and if made 0 from left the size of the button will decrease

Comment (Mahmoud-Bassuni) [reply Shared_UI/Shared_UI/Extension/UITextField + Extension.swift:27]: did got it

Comment (AhmedAtef277) [reply Shared_UI/Shared_UI/Extension/UITextField + Extension.swift:27]: got what?

**Review by Mahmoud-Bassuni — APPROVED** [review]

**Review by Mahmoud-Bassuni — APPROVED** [review]

**Review by ZeyadSultan — APPROVED** [review]

**Review by hassanSayed0 — APPROVED** [review]
